python_modules/libraries/dagster-databricks/dagster_databricks/databricks.py

```diff
@@ -35,23 +37,50 @@ def __init__(self, host: str, token: str, workspace_id: Optional[str] = None):
         self.host = host
         self.workspace_id = workspace_id
 
+        self._workspace_client = WorkspaceClient(
+            host=host,
+            token=token,
+            product="dagster-databricks",
+            product_version=__version__,
+        )
+
         # TODO: This is the old shim client that we were previously using. Arguably this is
         # confusing for users to use since this is an unofficial wrapper around the documented
-        # Databricks REST API. We should consider removing this in the future.
-        self.client = DatabricksAPI(host=host, token=token)
-        self.__setup_user_agent(self.client.client)
-
-        # Expose an interface directly to the official Databricks API client.
-        self._api_client = ApiClient(host=host, token=token)
+        # Databricks REST API. We should consider removing this in the next minor release.
+        self._client = databricks_api.DatabricksAPI(host=host, token=token)
+        self.__setup_user_agent(self._client.client)
+
+        # TODO: This is the old `databricks_cli` client that was previous recommended by Databricks.
+        # It is no longer supported and should be removed in favour of `databricks-sdk` in the next
+        # minor release.
+        self._api_client = databricks_cli.sdk.ApiClient(host=host, token=token)
         self.__setup_user_agent(self._api_client)
 
-    def __setup_user_agent(self, client: ApiClient) -> None:
+    def __setup_user_agent(self, client: databricks_cli.sdk.ApiClient) -> None:
         """Overrides the user agent for the Databricks API client."""
         client.default_headers["user-agent"] = f"dagster-databricks/{__version__}"
 
+    @deprecated
+    @public
+    @property
+    def client(self) -> databricks_api.DatabricksAPI:
+        warnings.warn(
+            (
+                "The `databricks_api.DatabricksAPI` client is deprecated and will be removed in a"
+                " future release. Please use `DatabricksClient.workspace_client`."
+            ),
+            DeprecationWarning,
+        )
+        return self._client
+
+    @client.setter
+    def client(self, value: databricks_api.DatabricksAPI) -> None:
+        self._client = value
```

Comment:
```suggestion
```

Reply:
- Why do we want to remove the warnings? The [docstring](https://github.com/dagster-io/dagster/blob/3f9ea13096ec551a69cae5a542781a2c5a433ecc/python_modules/dagster/dagster/_annotations.py#L59) for the `deprecated` decorator specifically says that warnings should be added separately.
- If we don't have the setter for the `client` property we've introduced a breaking change since the user used to be able to assign to `.client` (even if they shouldn't have!).